src/main/drivers/pwm_rx.h

```diff
@@ -17,6 +17,8 @@
 
 #pragma once
 
+#include "drivers/timer.h"
```

Comment:
Yes, but it will be noticed by the build process and is easily fixed.

The reason I favour forward declarations is that it reduces the number of includes, sometimes quite considerably. In total this has a significant impact on build times.

So the choices are:

1. Use, forward defs and only get name changes noticed at link time. But have shorter build times, so that it is feasible to do `make all` every now and then.
2. Don't use forward definitions, so name changes get noticed earlier. But other build problems don't get noticed because `make all` takes so long that noone ever does it.